core/src/sumeragi/mod.rs

```diff
@@ -93,25 +104,71 @@ impl Sumeragi {
                 telemetry_started,
                 transaction_limits: configuration.transaction_limits,
                 transaction_validator,
-                genesis_network,
                 queue,
                 broker,
                 kura,
-                // network,
+                network,
                 actor_channel_capacity: configuration.actor_channel_capacity,
                 fault_injection: PhantomData,
                 gossip_batch_size: configuration.gossip_batch_size,
                 gossip_period: Duration::from_millis(configuration.gossip_period_ms),
+
+                atomic_sumeragi_state: Mutex::new(atomic_sumeragi_state),
```

Comment:
From experience, whenever a Rust programmer sees a `Mutex`, they are itching to optimise it to an `RwLock`. Given they see `atomic` they might get rid of the `Mutex`. 

If it's a bad idea, you should explain it in a comment.

Reply:
I will do so.